specs/default/chef/site-cookbooks/hpcpack/files/InstallUtilities.psm1

```diff
@@ -111,12 +112,16 @@ function Get-MsiAccessToken
     Param
     (
         [parameter(Mandatory = $true)]
-        [string] $Resource
+        [string] $Resource,
+
+        [parameter(Mandatory = $true)]
+        [string] $ManagedId
     )
-    
+
     $Resource = $Resource.Trim()
     $encodedResource = [uri]::EscapeDataString($Resource)
-    $tokenUri = "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=$encodedResource"
+    $encodedManagedId = [uri]::EscapeDataString($ManagedId)
+    $tokenUri = "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=$encodedResource&msi_res_id=$encodedManagedId"
     $resp = Invoke-WebRequest -Uri $tokenUri -Method Get -Headers @{Metadata="true"} -UseBasicParsing
```

Comment:
shall we explicitly specify 'No proxy' option here?

Reply:
Sure